mesonbuild/build.py

```diff
@@ -637,13 +637,17 @@ def construct_id_from_path(subdir: str, name: str, type_suffix: str) -> str:
             return subdir_part + '@@' + my_id
         return my_id
 
-    def get_id(self) -> str:
+    @lazy_property
+    def __id(self) -> str:
```

Comment:
Just curious: why two leading underscores?

Reply:
It prevents it from being used unintentionally outside the definition of the class: the result of lazy_property is assigned to `__BuildTarget_id` instead of `__id`.

I did this to ensure that the code remains consistent instead of being a mess of `get_id()` and `id`. Either people keep using `get_id()`, or alternatively _all_ uses of `get_id()` can be converted to `id` at the same time that the underscores are removed.